common/include/abstract_protocol.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * abstract_protocol.h
+ *
+ *  Created on: 2020��9��30��
+ *      Author: duanning
+ */
+
+#ifndef INCLUDE_ABSTRACT_PROTOCOL_H_
+#define INCLUDE_ABSTRACT_PROTOCOL_H_
+
+#include <vector>
+#include <stdint.h>
+#include "rservice.h"
+
+class AbstractProtocol {
+public:
+
+	virtual int PackFrame() {};
+	virtual bool UnpackFrame() {};
+	virtual bool IsFrameComplete() {};
+	virtual bool IsFrameValid() {};
+
+private:
+	virtual int CalcCRC(vector<char>);
```

Comment:
An "abstract" class should not have "private" members, if we do need it, please make it public. Private virtual member functions will not be reloaded.

Reply:
Got it

Reply:
Please make this function public and use `uint8_t` to replace `char`.